File: moreapps/src/main/java/com/rocky/moreapps/model/HardUpdateDetails.java
package com.rocky.moreapps.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class HardUpdateDetails implements Parcelable {
    @SerializedName("enable")
    public boolean enable;

    @SerializedName("dialog_title")
    public String dialogTitle;

    @SerializedName("dialog_message")
    public String dialogMessage;

    @SerializedName("positive_button")
    public String positiveButton;

    protected HardUpdateDetails(Parcel in) {
        enable = in.readByte() != 0;
        dialogTitle = in.readString();
        dialogMessage = in.readString();
        positiveButton = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeByte((byte) (enable ? 1 : 0));
        dest.writeString(dialogTitle);
        dest.writeString(dialogMessage);
        dest.writeString(positiveButton);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<HardUpdateDetails> CREATOR = new Creator<HardUpdateDetails>() {
        @Override
        public HardUpdateDetails createFromParcel(Parcel in) {
            return new HardUpdateDetails(in);
        }

        @Override
        public HardUpdateDetails[] newArray(int size) {
            return new HardUpdateDetails[size];
        }
    };
}

File: moreapps/src/main/java/io/github/raghavsatyadev/moreapps/model/HardUpdateDetails.kt
package io.github.raghavsatyadev.moreapps.model

import android.os.Parcelable
import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Keep
@Parcelize
data class HardUpdateDetails(
    @SerializedName("enable")
    var enable: Boolean,

    @SerializedName("dialog_title")
    var dialogTitle: String,

    @SerializedName("dialog_message")
    var dialogMessage: String,

    @SerializedName("positive_button")
    var positiveButton: String,
) : Parcelable